app/post_metadata.go

```diff
@@ -53,11 +53,35 @@ func (a *App) PreparePostListForClient(originalList *model.PostList) *model.Post
 	return list
 }
 
+// PostWithOverridenIconURL changes the post icon override URL prop, if it has an emoji icon,
+// so that it points to the URL (relative) of the emoji - static if emoji is default, /api if custom.
+func (a *App) PostWithOverridenIconURL(post *model.Post) *model.Post {
+	prop, ok := post.Props[model.POST_PROPS_OVERRIDE_ICON_EMOJI]
+	if !ok || prop == nil {
+		return post
+	}
+	emojiName := prop.(string)
+	delete(post.Props, model.POST_PROPS_OVERRIDE_ICON_EMOJI)
+
+	if !*a.Config().ServiceSettings.EnablePostIconOverride || emojiName == "" {
+		return post
```

Comment:
nit: Probably also copy-paste, but seems weird to return `post` when we're just modifying it in place. I'd suggest either cloning or just skipping the return parameter.

Reply:
It started as a clone but the current clone is shallow. I kept the return to keep a more functional interface even if not really, but I agree, it's just confusing.